### DISCUSSION TITLE
Code snippet doesnt work in markdown

### DISCUSSION THREAD

**Discussion by xkeshav** [microsoft/vscode-discussions#1360] [Extension Development QnA] [answered]
I was trying to develop a snippet extension  but before that I wanted to run in the local code snippet method

So create user custom user defined code snippets way for markdown file but it doesnt work for markdown file 

## Approach 1: using local .code-snippets way

> F1 > Configure User Snippet  > New Snippet file for *project* name
and  this file stored in project root **/.vscode** location

### project/.vscode/post.code-snippets

```json
{
	"blog front matter": {
                "scope": "markdown, plantext",
		"prefix": "bfm",
		"body": [
			"---",
			"\ntitle: \"${1:title}\"",
			"publishDate: \"$CURRENT_DATE $CURRENT_MONTH_NAME_SHORT $CURRENT_YEAR $CURRENT_HOUR:$CURRENT_MINUTE $CURRENT_TIMEZONE_OFFSET\"",
			"\n---",
			"$0"
		],
		"description": "Creates an front matter boilarplate for astro blog"
	}
}
```

by changings `scope: "plaintext" ` then it is working as expected for plaintext file

## Approach 2: using **markdown.json**

> F1 > Configure User Snippet  > and selecting markdown in the list ; which create **.json** file and stored in vs code 

and no `scope` key here required [as per documentation ](https://code.visualstudio.com/docs/editor/userdefinedsnippets)

### ~/.config/Code/User/snippets/markdown.json

```json
{
	"blog front matter": {
		"prefix": "bfm",
		"body": [
			"---",
			"\ntitle: \"${1:title}\"",
			"publishDate: \"$CURRENT_DATE $CURRENT_MONTH_NAME_SHORT $CURRENT_YEAR $CURRENT_HOUR:$CURRENT_MINUTE $CURRENT_TIMEZONE_OFFSET\"",
			"\n---",
			"$0"
		],
		"description": "Creates an front matter boilarplate for astro blog"
	}
}

```

but it is not working at all

also having this settings enabled in workspace settings

```json
"editor.snippetSuggestions": "top",
```

what is wrong here; seen there are some question which says it doesnt support?

**Comment by ArturoDent** [comment]
See https://stackoverflow.com/a/71589996/836330 and the comment at https://stackoverflow.com/questions/64134211/can-we-write-code-snippet-for-markdown-language-in-visual-studio-code.

**Reply by xkeshav** [reply]
Thanks I am able to use the `.code-snippet` file ( which has scope key) 

by adding suggested settings in _.code-workspace_ file. 

```json
 "[markdown]": {
      "editor.snippetSuggestions": "top",
      "editor.defaultFormatter": "yzhang.markdown-all-in-one",
      "editor.quickSuggestions": {
        "other": true,
        "comments": true,
        "strings": true
      }
    },
```

**Reply by robconery** [reply, marked as answer]
Great - does this solve the issue?

**Comment by robconery** [comment]
I _think_ one of the reasons might be that Markdown isn't code, so there's no intellisense trigger for the snippet. Does your snippet work with `CTRL-space`? I just tried on my end and it worked.

**Comment by xkeshav** [comment]
Yes. It worked. Thanks
